gcc/rust/expand/rust-expand-visitor.cc

```diff
@@ -1264,12 +1264,13 @@ ExpandVisitor::visit (AST::ExternalFunctionItem &item)
   for (auto &param : item.get_generic_params ())
     visit (param);
 
-  // FIXME: Should this work? What is the difference between NamedFunctionParam
-  // and FunctionParam?
-  // expand_function_params (item.get_function_params ());
+  auto end = item.is_variadic () ? item.get_function_params ().end () - 1
```

Comment:
This seems sus... are you trying to get the last argument?

Reply:
I'm trying to avoid the last argument in the for loop if the function is variadic because this argument has no type. I recognize it is not as easy to read as I would like but I haven't found a better way to express it.

Reply:
Now that I've worked on #2713 I know that AST validation pass is done after the expansion dump. This means there may be some invalid variadic function as shown in issues #2710 and #2711 going in that loop whose type cannot be fetched. Instead of avoiding the last parameter when the function is variadic, we should check for the presence of a type in the loop.

Reply:
Rebased and fixed!